### PR TITLE
Restructure example/trl/stack_llama_2 for generic DPO 

### PR DESCRIPTION
**Pull request by libinta**
Restructure example/trl/stack_llama_2 for generic DPO. Will add Anthropic dataset later

# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/trl/README.md [hunk 1]
```diff
@@ -8,18 +9,21 @@ Install all the dependencies in the `requirements.txt`:
 $ pip install -U -r requirements.txt
 ```
 
+## DPO pipeline
 
-## Training
+### Training
 
+The following example is for the creation of StackLlaMa 2: a Stack exchange llama-v2-7b model
 There were two main steps to the DPO training process:
```
Comment (regisss) [inline examples/trl/README.md:17]:
```suggestion
The following example is for the creation of StackLlaMa 2: a Stack exchange llama-v2-7b model.
There are two main steps to the DPO training process:
```

**Review by regisss — COMMENTED** [review]
LGTM! You'll need to run `make style` too.

**Comment by libinta** [comment]
> LGTM! You'll need to run `make style` too.

done

**Review by sywangyi — APPROVED** [review]
LGTM

File: examples/trl/README.md [hunk 2]
```diff
@@ -28,23 +32,27 @@ There were two main steps to the DPO training process:
...
+        --lora_target_modules "q_proj" "v_proj" \
         --bf16 \
         --remove_unused_columns=False \
         --run_name="sft_llama2" \
         --report_to=none \
         --use_habana \
-        --use_lazy_mode
+        --use_lazy_mode \
+
     ```
```
Comment (regisss) [inline examples/trl/README.md:43]:
```suggestion
        --use_lazy_mode
    ```
```

File: examples/trl/README.md [hunk 3]
```diff
@@ -54,19 +62,14 @@ python merge_peft_adapter.py --base_model_name="meta-llama/Llama-2-7b-hf" --adap
...
 which will also push the model to your HuggingFace hub account.
 
-## Running the model
-
-We can load the DPO-trained LoRA adaptors which were saved by the DPO training step and load them via:
+### Running the model
 
-```py
-from peft import AutoPeftModelForCausalLM
+We can load the DPO-trained LoRA adaptors which were saved by the DPO training step and run it through example/text-generation folder.
```
Comment (regisss) [inline examples/trl/README.md:67]:
```suggestion
We can load the DPO-trained LoRA adaptors which were saved by the DPO training step and run it through the [text-generation example](https://github.com/huggingface/optimum-habana/tree/main/examples/text-generation).
```

**Review by regisss — COMMENTED** [review]
I left a couple of comments that won't take long to address

File: examples/trl/README.md [hunk 4]
```diff
@@ -28,23 +32,26 @@ There were two main steps to the DPO training process:
...
         --weight_decay=0.05 \
         --optim="paged_adamw_32bit" \
+        --lora_target_modules "q_proj" "v_proj" \
         --bf16 \
         --remove_unused_columns=False \
         --run_name="sft_llama2" \
         --report_to=none \
         --use_habana \
-        --use_lazy_mode
+        --use_lazy_mode \
```
Comment (regisss) [inline examples/trl/README.md:41]:
```suggestion
        --use_lazy_mode
```

**Review by regisss — APPROVED** [review]

**Pull request merged**